src/Lean/Elab/MutualDef.lean

```diff
@@ -824,10 +824,10 @@ private partial def mkClosureForAux (toProcess : Array FVarId) : StateRefT Closu
     | .cdecl _ _ userName type bi k =>
       let toProcess ← pushLocalDecl toProcess fvarId userName type bi k
       mkClosureForAux toProcess
-    | .ldecl _ _ userName type val _ k =>
+    | .ldecl _ _ userName type val nondep k =>
       let zetaDeltaFVarIds ← getZetaDeltaFVarIds
-      if !zetaDeltaFVarIds.contains fvarId then
-        /- Non-dependent let-decl. See comment at src/Lean/Meta/Closure.lean -/
+      if nondep || !zetaDeltaFVarIds.contains fvarId then
+        /- Nondependent let-decl. See comment at src/Lean/Meta/Closure.lean -/
```

Comment:
@kmill  This is an optimization, correct? I am assuming that if `fvarId` is in `zetaDeltaFVarIds` and `nondep` is `true`, then the `nondep` annotation is incorrect.

Reply:
Correct, if `nondep` is true then it is not possible for `zetaDeltaFVarIds` to contain `fvarId`. This is just a small optimization.

Reply:
Added a comment mentioning this invariant.